src/frontend/src/components/PasswordReset/index.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { AppState } from '../../redux/store';
+import { logIn } from '../../redux/system/actions';
+import { SystemState } from '../../redux/system/types';
+import Logo from '../Logo';
+import history from '../History';
+
+interface Props {
+  system: SystemState;
+  logIn: typeof logIn;
+}
+interface State {
+  password: string;
+  missingPasswordInput: null | boolean;
+  passwordIsValid: null | boolean;
+}
+
+class PasswordReset extends React.Component<Props, State> {
+  state: State = {
+    password: '',
+    missingPasswordInput: null,
+    passwordIsValid: null
+  };
+
+  handleChange = (e: React.BaseSyntheticEvent) => {
+    // See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/26635 for why we're
+    // using the "any" type.
+    this.setState<any>({
+      [e.target.id]: e.target.value
+    });
+  };
+
+  handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    this.validateForm();
+  };
+
+  validateForm = () => {
+    this.setState({
+      missingPasswordInput: this.state.password.trim().length === 0
+    });
+    this.setState(
+      {
+        passwordIsValid: this.state.password.trim().length >= 10 // Need to implement a more robust password check.
+      },
+      () =>
+        this.state.passwordIsValid === true &&
+        this.state.missingPasswordInput === false
+          ? history.push('/')
+          : null // Using setState(updater, callback) beccause setState doesn't immediately update component
+    );
```

Comment:
Same comment as above, this side effect (state change) should be in handleSubmit, not as a side effect of validation. Validation should validate only and do nothing else.

Reply:
Using `<Redirect>` as above. Moved out of validation.